core/engine/src/object/jsobject.rs

```diff
@@ -287,6 +287,42 @@ impl JsObject {
         std::ptr::eq(self.vtable(), &ARRAY_EXOTIC_INTERNAL_METHODS)
     }
 
+    /// Checks that all own property keys and values are equal (recursively).
+    #[must_use]
+    #[inline]
+    pub fn deep_strict_equals(lhs: &Self, rhs: &Self, context: &mut Context) -> bool {
+        let Ok(l_keys) = lhs.own_property_keys(context) else {
+            return false;
+        };
+        let Ok(r_keys) = lhs.own_property_keys(context) else {
+            return false;
+        };
```

Comment:
I think it's best to return a `JsResult<bool>` and propagate the error, since calling the `own_property_keys` could have side effects.